### PR TITLE
Cleaned USE_PTHREAD codebase as requested in #147

### PR DESCRIPTION
**Pull request by st4rl3ss**
Cleaned all instances of USE_PTHREAD.

### CODE REVIEW

File: coreneuron/nrnoc/multicore.cpp [hunk 1]
```diff
@@ -459,27 +232,6 @@ void nrn_multithread_job(void* (*job)(NrnThread*)) {
...
-        nrn_inthread_ = 0;
-    } else { /* sequential */
-#else
-    {
-#endif
-        for (i = 1; i < nrn_nthread; ++i) {
-            (*job)(nrn_threads + i);
-        }
-        (*job)(nrn_threads);
-    }
```
Comment (pramodk) [inline coreneuron/nrnoc/multicore.cpp]: Could you verify if you haven't deleted too much here?

**Comment by pramodk** [comment]
When I build this on Mac (with clang compiler), I see:

```
[ 20%] Building CXX object coreneuron/CMakeFiles/coreneuron.dir/nrniv/cvodestb.cpp.o
cd /Users/kumbhar/workarena/repos/bbp/coreneuron/build/coreneuron && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -DCORENEURON_BUILD -DDISABLE_HOC_EXP -DENABLE_SPLAYTREE_QUEUING -DLAYOUT=0 -DNRNMPI=1 -Dcoreneuron_EXPORTS -I/Users/kumbhar/workarena/repos/bbp/coreneuron/coreneuron/utils/randoms -I/Users/kumbhar/workarena/repos/bbp/coreneuron/build/coreneuron -I/Users/kumbhar/workarena/repos/bbp/coreneuron -I/Users/kumbhar/workarena/repos/bbp/coreneuron/coreneuron -I/usr/local/Cellar/open-mpi/4.0.0/include  -g  -O2 -fPIC   -std=c++11 -o CMakeFiles/coreneuron.dir/nrniv/cvodestb.cpp.o -c /Users/kumbhar/workarena/repos/bbp/coreneuron/coreneuron/nrniv/cvodestb.cpp
In file included from /Users/kumbhar/workarena/repos/bbp/coreneuron/coreneuron/nrniv/cvodestb.cpp:33:
In file included from /Users/kumbhar/workarena/repos/bbp/coreneuron/coreneuron/nrniv/netcvode.h:32:
/Users/kumbhar/workarena/repos/bbp/coreneuron/coreneuron/nrniv/tqueue.h:210:5: error: unknown type name 'MUTDEC'
    MUTDEC
    ^
/Users/kumbhar/workarena/repos/bbp/coreneuron/coreneuron/nrniv/tqueue.h:215:1: error: expected member name or ';' after declaration specifiers
};
^
In file included from /Users/kumbhar/workarena/repos/bbp/coreneuron/coreneuron/nrniv/cvodestb.cpp:33:
In file included from /Users/kumbhar/workarena/repos/bbp/coreneuron/coreneuron/nrniv/netcvode.h:32:
In file included from /Users/kumbhar/workarena/repos/bbp/coreneuron/coreneuron/nrniv/tqueue.h:217:
/Users/kumbhar/workarena/repos/bbp/coreneuron/coreneuron/nrniv/tqueue.ipp:58:5: error: use of undeclared identifier 'MUTCONSTRUCT'
    MUTCONSTRUCT(0)
    ^
/Users/kumbhar/workarena/repos/bbp/coreneuron/coreneuron/nrniv/tqueue.ipp:93:5: error: use of undeclared identifier 'MUTDESTRUCT'
    MUTDESTRUCT
    ^
/Users/kumbhar/workarena/repos/bbp/coreneuron/coreneuron/nrniv/tqueue.ipp:98:5: error: unknown type name 'MUTLOCK'
```

> P.S. Issue is obvious because OS X Clang doesnt have OpenMP.

File: coreneuron/nrnoc/multicore.cpp (hunk 1 above)
Comment (st4rl3ss) [reply coreneuron/nrnoc/multicore.cpp]: I deleted too much indeed! :)

**Comment by st4rl3ss** [comment]
This should fix the issues in building without OPENMP and MPI.
Also restored part of the code that was wrongly deleted. Build and test run correctly on my local machine with or without OPENMP and MPI.

**Comment by pramodk** [comment]
Please retest

**Comment by pramodk** [comment]
@st4rl3ss  : check why Jenkins pipeline is failing. @iomaganaris has setup that.

**Comment by iomaganaris** [comment]
> @st4rl3ss : check why Jenkins pipeline is failing. @iomaganaris has setup that.

After some recent changes the pipeline fails but I haven't figured out why yet. I will try to disable it from PRs up until it is sure that it works fine. I am sorry for the inconvenience.

**Comment by iomaganaris** [comment]
Please retest

**Comment by st4rl3ss** [comment]
please retest

**Comment by st4rl3ss** [comment]
Please retest

**Comment by st4rl3ss** [comment]
Please retest

**Comment by iomaganaris** [comment]
please retest

**Comment by iomaganaris** [comment]
please retest

**Comment by st4rl3ss** [comment]
please retest

**Comment by iomaganaris** [comment]
please retest

File: coreneuron/nrniv/nrn_setup.cpp [hunk 2]
```diff
@@ -682,8 +687,7 @@ void nrn_setup(const char* filesdat,
     // Fortunately, empty threads work fine.
     // Allocate NrnThread* nrn_threads of size ngroup (minimum 2)
     // Note that rank with 0 dataset/cellgroup works fine
-    nrn_threads_create(ngroup <= 1 ? 2 : ngroup,
-                       nrnopt_get_flag("--threading") ? 1 : 0);  // serial/parallel threads
+    nrn_threads_create(ngroup <= 1 ? 2 : ngroup);  // serial/parallel threads
```
Comment (pramodk) [inline coreneuron/nrniv/nrn_setup.cpp]: `// serial/parallel threads` comment is irrelevant now

**Pull request merged**